lua/gitsigns/git.lua

```diff
@@ -231,8 +231,17 @@ function Obj.new(file, revision, encoding, gitdir, toplevel)
     return
   end
 
-  if not vim.startswith(file, '/') and toplevel then
-    file = toplevel .. util.path_sep .. file
+  if toplevel then
+    -- windows
+    if vim.fn.has('win32') then
+      if not string.match(file, '^[A-Za-z]:[/\\]') then
+        file = toplevel .. util.path_sep .. file
+      end
+
+    -- unix
+    elseif not vim.startswith(file, '/') then
+      file = toplevel .. util.path_sep .. file
+    end
```

Comment:
I don't agree with this change. You are special casing when `file` starts with a drive AND is on Windows, and haven't explained it.

```suggestion
if not vim.startswith(file, '/') and toplevel then
  file = toplevel .. util.path_sep .. file
end

file = util.normalize_path(file)
```

And move `normalize_path` from `repo.lua` to `util.lua`.

Reply:
I'm not sure this change will be necessary if we change the commands to use the relative path.

This will also need the helper function for better absolute path detection I added in the other PR.